src/radical/pilot/agent/executing/base.py

```diff
@@ -341,7 +343,8 @@ def _create_exec_script(self, launcher, task):
             tmp += self._separator
             tmp += '# output file detection (ii)\n'
             tmp += 'ls | sort | comm -23 - ' \
-                   "%s.files | grep -ve '^%s\\.' > %s.ofiles\n" % (tid, tid, tid)
+                   "%s.files | grep -ve '^%s\\.' > %s/%s.ofiles\n" \
```

Comment:
`%s.files` it should be `'%s/%s.files' % (env_sbox, tid)`
(we pick `.files`-file from task sandbox)

p.s., I also wondering if we can use `$RP_TASK_ID` instead of actual value

Reply:
Great catch - fixed!